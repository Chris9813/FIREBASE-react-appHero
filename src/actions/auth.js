import Swal from "sweetalert2"
import {firebase, googleAuthProvider} from "../firebase/firebase-config"
import { types } from "../types/types"


export const startLoginEmailPaswword = (email, password) => {
    return (dispatch) => {
        return firebase.auth().signInWithEmailAndPassword(email, password)
        .then(({user}) => {
            console.log(user);
            dispatch(login(user.uid, user.displayName))
        }).catch(e => {
            console.log(e);
            Swal.fire("Error", e.message, "error")
        })
        
    }
}


export const startGoogleLogin = () =>{
    return (dispatch) =>{
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( userCred =>  {
                console.log(userCred);
            })
    }
}


export const login = (uid, displayname) => ({
    type: types.login,
    payload: {
        uid,
        displayname,
    }
})

export const startLogout = () => {
    return async(dispatch) => {
        await firebase.auth().signOut()
        dispatch(logout())
    }
}

export const logout = () => ({
    type: types.logout
})


