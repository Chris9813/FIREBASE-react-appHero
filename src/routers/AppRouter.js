import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
    import {
        BrowserRouter as Router,
        Switch
    } from "react-router-dom";
import { LoginScreen } from '../components/login/LoginScreen';
import { DeshboardRoutes } from "./DeshboardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
import {firebase} from '../firebase/firebase-config'
import { login } from "../actions/auth";

export const AppRouter = () => {

    const dispatch = useDispatch()

    const [checking, setchecking] = useState(true);
    const [isLoggedIn, setisLoggedIn] = useState(false)


    useEffect(() => {
        firebase.auth().onAuthStateChanged(async(user) =>{
            if(user?.uid) {
                dispatch(login(user.uid, user.displayName))
                setisLoggedIn(true)
            } else {
                setisLoggedIn(false)
            }
            setchecking(false)
        })
    }, [dispatch, setchecking, setisLoggedIn])


if(checking) {
    return(
        <h1>Wait...</h1>
    )
}

    return (
        <div>
        <Router>
    <div>
        <div>
        <Switch>
        <PublicRoute
        exact path = "/login" component = {LoginScreen}
        isAuthenticated = {isLoggedIn}
        />

        <PrivateRoute
        path="/" 
        component = {DeshboardRoutes} 
        isAuthenticated = {isLoggedIn}
        />

        </Switch>
        </div>
        
    </div>
    </Router>
        </div>
        
    )
}
