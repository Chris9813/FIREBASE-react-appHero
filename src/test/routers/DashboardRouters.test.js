import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../auth/AuthContext";
import { DeshboardRoutes } from "../../routers/DeshboardRoutes";

describe('Pruebas en <DasboardRoutes />', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: "CHRIS"
        }
    }

    
    test('debe mostrarse correctamente ', () => {

        const wrapper = mount(
            <MemoryRouter>
                <AuthContext.Provider value = {contextValue}>
                <DeshboardRoutes />
            </AuthContext.Provider>
            </MemoryRouter>
            
        )
        expect(wrapper).toMatchSnapshot();
    })


})
