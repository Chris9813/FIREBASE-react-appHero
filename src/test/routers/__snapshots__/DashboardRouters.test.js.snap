// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DasboardRoutes /> debe mostrarse correctamente  1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info align-self-end"
        />
      </ul>
    </div>
    <div
      className="navbar-collapse collapse w-200 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="container"
      >
        <div
          className="container"
        >
          <div
            className="row row-cols-3 animate__animated animate__backInDown"
          >
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      X-Men #1
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="row row-cols-2"
                style={
                  Object {
                    "maxWidth": 540,
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    alt="{superhero}"
                    className="card-img-top"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text-center"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small>
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    className="link-info"
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
